Clarify concurrency cancel-in-progress behaviour (#30647)
github/docs: data/reusables/actions/actions-group-concurrency.md

> When a concurrent job or workflow is queued, if another job or workflow using the same concurrency group in the repository is in progress, the queued job or workflow will be `pending`. Any pending job or workflow in the concurrency group will be canceled.  This means that there can be at most 1 running and 1 pending job in a concurrency group at any one time.

Reviewer: ```suggestion
When a concurrent job or workflow is queued, if another job or workflow using the same concurrency group in the repository is in progress, the queued job or workflow will be `pending`. Any pending job or workflow in the concurrency group will be canceled.  This means that there can be at most one running and one pending job in a concurrency group at any time.
```
Just updating to match our style guide (write out numbers less than 10).

> To also cancel any currently running job or workflow in the same concurrency group, specify `cancel-in-progress: true`.  You may also specify `cancel-in-progress` as an expression subject to the restrictions on available contexts outlined above.

Reviewer: ```suggestion
To also cancel any currently running job or workflow in the same concurrency group, specify `cancel-in-progress: true`.  To conditionally cancel currently running jobs or workflows in the same concurrency group, you can specify `cancel-in-progress` as an expression if you use any of the following allowed expression contexts: [`github`](/actions/learn-github-actions/contexts#github-context), [`inputs`](/actions/learn-github-actions/contexts#inputs-context), [`vars`](/actions/learn-github-actions/contexts#vars-context), [`needs`](/actions/learn-github-actions/contexts#needs-context), [`strategy`](/actions/learn-github-actions/contexts#strategy-context), and [`matrix`](/actions/learn-github-actions/contexts#matrix-context).
```
We try to refrain from saying things like "outlined above" in reusables, because if the reusable is placed in a doc where the thing isn't outlined above, that would be really confusing. So I just repeated the information here.

> To also cancel any currently running job or workflow in the same concurrency group, specify `cancel-in-progress: true`.  To conditionally cancel currently running jobs or workflows in the same concurrency group, you can specify `cancel-in-progress` as an expression if you use any of the following allowed expression contexts: [`github`](/actions/learn-github-actions/contexts#github-context), [`inputs`](/actions/learn-github-actions/contexts#inputs-context), [`vars`](/actions/learn-github-actions/contexts#vars-context), [`needs`](/actions/learn-github-actions/contexts#needs-context), [`strategy`](/actions/learn-github-actions/contexts#strategy-context), and [`matrix`](/actions/learn-github-actions/contexts#matrix-context).

Reviewer: ```suggestion
To also cancel any currently running job or workflow in the same concurrency group, specify `cancel-in-progress: true`.  To conditionally cancel currently running jobs or workflows in the same concurrency group, you can specify `cancel-in-progress` as an expression with any of the allowed expression contexts.
```
It was too long repeating specific information, so I'm updating it to this.

```

{% endraw %}

### Example: Only cancel in-progress jobs on specific branches

>  In some scenarios it may be desireable to cancel in-progress jobs on certain branches but not others e.g. development versus release branches.  You can use conditional expressions with `cancel-in-progress` to achieve this.

Reviewer: ```suggestion
If you would like to cancel in-progress jobs on certain branches but not on others, you can use conditional expressions with `cancel-in-progress`. For example, you can do this if you would like to cancel in-progress jobs on development branches but not on release branches.
```
Just editing the style to match our content. 👍

> To only cancel in-progress runs of the same workflow when not running on a release branch you could do the following:

Reviewer: ```suggestion
To only cancel in-progress runs of the same workflow when not running on a release branch, you can set `cancel-in-progress` to an expression similar to the following:
```

{% raw %}

```yaml
concurrency:
  group: ${{ github.workflow }}-${{ github.ref }}
  cancel-in-progress: ${{ !contains(github.ref, 'release/')}}
```

{% endraw %}

> So multiple pushes to a `release/1.2.3` branch would not cancel in-progress runs, whereas pushes to another branch e.g. `main` would cancel in-progress runs.

Reviewer: ```suggestion
In this example, multiple pushes to a `release/1.2.3` branch would not cancel in-progress runs. Pushes to another branch, such as `main`, would cancel in-progress runs.
```
